lib/api/core/sdk/funnelProtocol.js

```diff
@@ -22,23 +22,44 @@
 'use strict';
 
 const
-  KuzzleEventEmitter = require('kuzzle-sdk').KuzzleEventEmitter,
-  Request = require('kuzzle-common-objects').Request;
+  {
+    Request,
+    errors: {
+      InternalError: KuzzleInternalError
+    }
+  } = require('kuzzle-common-objects'),
+  User = require('../models/security/user'),
+  KuzzleEventEmitter = require('kuzzle-sdk').KuzzleEventEmitter;
 
 class FunnelProtocol extends KuzzleEventEmitter {
-  constructor(funnel) {
+  constructor(funnel, user = null) {
     super();
+
     this.funnel = funnel;
     this.id = 'funnel';
+    this.user = user;
+
+    if (this.user && !(this.user instanceof User)) {
+      throw new KuzzleInternalError('FunnelProtocol.constructor: Invalid User object');
+    }
   }
 
-  isReady() {
+  isReady () {
     return true;
   }
 
+  /**
+   *  Hydrate the user and execute SDK query
+   */
   query (request) {
-    return this.funnel.executePluginRequest(new Request(request))
-      .then(result => ({result}));
+    const kuzzleRequest = new Request(request);
```

Comment:
Suggestion: maybe we could also inject a hard-coded `connection` objet to the context (for example with attribute `{protocol: 'funnel'}`)?
I mean construct the `Request` like this:
```js
    const kuzzleRequest = new Request(request, {
      connection: {protocol: "funnel"}
    });
```

Reply:
Good idea!